src/diffusers/configuration_utils.py

```diff
@@ -104,9 +136,98 @@ def save_config(self, save_directory: Union[str, os.PathLike], push_to_hub: bool
         logger.info(f"Configuration saved in {output_config_file}")
 
     @classmethod
-    def from_config(cls, pretrained_model_name_or_path: Union[str, os.PathLike], return_unused_kwargs=False, **kwargs):
+    def from_config(cls, config: Union[FrozenDict, Dict[str, Any]] = None, return_unused_kwargs=False, **kwargs):
```

Comment:
Why the distinction between `FrozenDict` and `Dict` instead of just `Dict`?

Reply:
I do understand that, it's just that `Union[FrozenDict, Dict]` is the same as `Dict`, isn't it?

Reply:
Yeah pretty much :-)